test_scripts/Defects/8_1/994_SDL_sends_appName_in_vrSynonyms_and_ttsName_in_case_of_separated_structure_in_json_file.lua

```diff
@@ -0,0 +1,146 @@
+    if item.name == pAppData.appName then
+      isAppFound = true
+      local deviceInfo
+      if item.ios ~= nil then
+        deviceInfo = item.ios
+      else
+        deviceInfo = item.android
+      end
+      local exp = {}
+      for _, tab in pairs(deviceInfo.languages) do
+        for l, params in pairs(tab) do
+          if l == "default" then
+            for p,v in pairs(params) do
+              if not exp[p] then
+                exp[p] = v
+              end
+            end
+          elseif l == "EN-US" then
+            for p,v in pairs(params) do
+              exp[p] = v
+            end
+          end
+        end
+      end
+      local act = pAppData
+      if exp.ttsName ~= act.ttsName[1] then
+        return false, "Expected 'ttsName' is '" .. tostring(exp.ttsName)
+          .. "', actual '" .. tostring(act.ttsName[1]) .. "'"
+      end
+      if not utils.isTableEqual(exp.vrSynonyms, act.vrSynonyms) then
+        return false, "Expected 'vrSynonyms' is:\n"  .. utils.tableToString(exp.vrSynonyms)
+          .. "\nactual:\n" .. utils.tableToString(act.vrSynonyms)
+      end
+    end
+  end
+  if isAppFound == false then
+    return false, "App wasn't found"
+  end
+  return true
+end
+
+local function registerAppWithOnSystemRequestQueryApps(pJsonFile)
+  local session = common.mobile.createSession()
+  session:StartService(7)
+  :Do(function()
+      local cid = session:SendRPC("RegisterAppInterface", common.getConfigAppParams())
+      common.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
+      session:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+      :Do(function()
+          common.getMobileSession():ExpectNotification("OnHMIStatus",
+          { hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+          :Do(function()
+              local msg = {
+                serviceType      = 7,
+                frameInfo        = 0,
+                rpcType          = 2,
+                rpcFunctionId    = 32768,
+                rpcCorrelationId = common.getMobileSession().correlationId + 30,
+                payload          = '{"hmiLevel":"FULL", "audioStreamingState":"NOT_AUDIBLE", "systemContext":"MAIN"}'
+              }
+              common.getMobileSession():Send(msg)
+            end)
+        end)
+    end)
+  common.getMobileSession():ExpectNotification("OnSystemRequest",
+    { requestType = "LOCK_SCREEN_ICON_URL" },
+    { requestType = "QUERY_APPS" })
+  :Times(2)
+  :Do(function(_, data)
+      if data.payload.requestType == "QUERY_APPS" then
+        local cid = common.getMobileSession():SendRPC("SystemRequest", {
+          requestType = "QUERY_APPS", fileName = pJsonFile },
+          pathToJsonFiles .. "/" .. pJsonFile)
+        common.getHMIConnection():ExpectRequest("BasicCommunication.UpdateAppList")
+        :ValidIf(function(_, data1)
+            for _, appData in ipairs(data1.params.applications) do
```

Comment:
This piece of code doesn't validate all required applications
It validates only the ones which are provided in `applications` array. 
On a current develop this list contains only default `"Test Application"` and misses apps provided in `json`
I would add an additional check on number of apps provided in `applications` array before checking parameters in `checkParams()` function